src/parser/ast/DefaultArgumentNode.h

```diff
@@ -0,0 +1,97 @@
+/*
+ * Copyright (c) 2016-present Samsung Electronics Co., Ltd
+ *
+ *  This library is free software; you can redistribute it and/or
+ *  modify it under the terms of the GNU Lesser General Public
+ *  License as published by the Free Software Foundation; either
+ *  version 2 of the License, or (at your option) any later version.
+ *
+ *  This library is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ *  Lesser General Public License for more details.
+ *
+ *  You should have received a copy of the GNU Lesser General Public
+ *  License along with this library; if not, write to the Free Software
+ *  Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301
+ *  USA
+ */
+
+#ifndef DefaultArgumentNode_h
+#define DefaultArgumentNode_h
+
+#include "ExpressionNode.h"
+#include "IdentifierNode.h"
+#include "PatternNode.h"
+#include "MemberExpressionNode.h"
+
+namespace Escargot {
+
+class DefaultArgumentNode : public ExpressionNode {
+public:
+    friend class ScriptParser;
+
+    DefaultArgumentNode(Node* left, Node* right, Context* ctx)
+        : ExpressionNode()
+        , m_left(left)
+        , m_right(right)
+        , m_ctx(ctx)
+    {
+    }
+
+    virtual ~DefaultArgumentNode()
+    {
+    }
+
+    void giveupChildren()
+    {
+        m_left = m_right = nullptr;
+    }
+
+    Node* left()
+    {
+        return m_left.get();
+    }
+
+    virtual ASTNodeType type() { return ASTNodeType::DefaultArgument; }
+    virtual AtomicString getParameterName()
+    {
+        return ((IdentifierNode*)m_left.get())->name();
+    }
+
+    virtual void generateResultNotRequiredExpressionByteCode(ByteCodeBlock* codeBlock, ByteCodeGenerateContext* context)
+    {
+        size_t undefinedIndex = context->getRegister();
+        codeBlock->pushCode(GetGlobalObject(ByteCodeLOC(m_loc.index), undefinedIndex, PropertyName(m_ctx->staticStrings().undefined)), context, this);
```

Comment:
`m_ctx` can be removed. We can access the `StaticStrings` through codeBlock->m_codeBlock->m_context.

Reply:
Ohh that is a good news! The only reason why I stored the `m_ctx` to access the static strings.